#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "pinger_tracker"

gen = ParameterGenerator()

gen.add("ADC_bits", int_t, 0, "ADC bits", 16, 2, 24)
gen.add("sample_rate", int_t, 0, "ADC sampling rate (kHz)", 300, 25, 2000)
gen.add("signal_rate", double_t, 0, "Transmission rate", 1.0, 0.1, 10.0)
gen.add("amplitude", double_t, 0, "Signal amplitude", 0.1, 0.01, 1.0)
gen.add("signal_freq", int_t, 0, "Signal Frequency (kHz)", 27, 25, 50)

exit(gen.generate(PACKAGE, "pinger_tracker", "Signal"))